mirrord-cli/src/main.rs

```diff
@@ -64,6 +64,83 @@ const INJECTION_ENV_VAR: &str = "LD_PRELOAD";
 #[cfg(target_os = "macos")]
 const INJECTION_ENV_VAR: &str = "DYLD_INSERT_LIBRARIES";
 
+/// This is for Apple M1 running mirrord aarch trying to execute x86 binaries.
+#[cfg(all(target_os = "macos", target_arch = "aarch64"))]
+mod mac_aarch {
+    use std::{
+        io::{Cursor, Read},
+        path::Path,
+    };
+
+    use mach_object::{OFile, CPU_TYPE_X86_64};
+    use reqwest;
+    use search_path::SearchPath;
+
+    use super::*;
+    pub fn is_binary_different_arch(binary_path: &String) -> bool {
+        let search_path = SearchPath::new("PATH").unwrap();
+        let binary_path = match search_path.find(&Path::new(binary_path)) {
```

Comment:
I would suggest combining these operations by using `and_then` and `map`, instead of having separate `match`es.

Reply:
but then I can't add appropriate logs?

Reply:
Think you still can do it.

Reply:
erm I'm still against it because as far as I see it, `and_then` and `map` are to avoid nesting, but if I'm just doing match -> return the code is "nicer" and has less nesting.

Reply:
Nope, these types of functions are not to avoid nesting, they exist to let you work in a "monadic context" (sorta) for as long as possible. What you're doing here is very similar to the `match => None` anti-pattern actually, I'm surprised clippy doesn't complain. [See here](https://github.com/rust-lang/rust-clippy/issues/6).

When you're dealing with mutation, `match`ing`, `for` loops and the other more imperative forms are better, but this is not case here.